CONTRIBUTE.md

```diff
@@ -0,0 +1,156 @@
+# Architecture
+
+Galaxykit has a fairly simple structure:
+
+```
+(galaxykit) henderson@mars => tree .
+.
+├── CHANGES.md
+├── CONTRIBUTE.md
+├── galaxykit
+│   ├── client.py
+│   ├── collections.py
+│   ├── command.py
+│   ├── container_images.py
+│   ├── containers.py
+│   ├── containerutils.py
+│   ├── groups.py
+│   ├── namespaces.py
+│   ├── registries.py
+│   └── users.py
+├── LICENSE.md
+├── README.md
+├── setup.cfg
+└── setup.py
+```
+
+Let's work through these files individually:
+
+### CHANGES.md
+A changelog, tracking all the updates to the code.
+
+### CONTRIBUTE.md
+
+This file, with advice on how to effectively contribute to galaxykit.
+
+### galaxykit:
+
+The source directory, containing all the code that goes into galaxykit.
+
+#### client.py
+
+This file contains the GalaxyClient object, which is essentially a single authenticated context for making requests against an existing galaxy_ng instance. This file and `command.py` contain the primary two interfaces to interacting with galaxykit.
+
+#### collections.py
+
+Functions for managing collections.
+
+#### command.py
+
+The command-line interface logic - this is what gets run when you invoke galaxykit directly on the command-line. Currently in a bit of a hacky state, as it's grown faster than we can refactor it. An extensive refactor is in progress - check out [this PR](https://github.com/ansible/galaxykit/pull/30) if you want to review the changes.
+
+#### container_images.py
+
+For interacting with individual container images.
+
+#### containers.py
+
+Functions for dealing with containers as as discrete units (deleting, setting metadata.)
+
+#### containerutils.py
+
+This code abstracts over common `podman` or `docker` commands for uploading or downloading container images from a galaxy_ng instance.
+
+#### groups.py
+
+Adding/deleting/modifying the permissions of groups.
+
+#### namespaces.py
+
+Creating and deleting namespaces.
+
+#### registries.py
+
+Adding new remote container registries.
+
+#### users.py
+
+Adding/deleting users.
+
+### LICENSE.md
+
+This is the GPLv2, which is the same licence galaxy_ng is distributed under.
+
+### README.md
+
+Landing page.
+
+### setup.cfg
```

Comment:
I think we should group these top-level files with CHANGES and CONTRIBUTION above